### PR TITLE
OCPBUGSM-28017: Unify operator manifests and move them under config

### PR DESCRIPTION
**Pull request by djzager**
This moves all of the operator manifests into the `config` directory,
redirects generation to use the `config` directory, and brings the OLM
manifests into `deploy/olm-catalog` for safe keeping.

Fixes:
[OCPBUGSM-28017](https://issues.redhat.com/browse/OCPBUGSM-28017)
[OCPBUGSM-28075](https://issues.redhat.com/browse/OCPBUGSM-28075)

### CODE REVIEW

File: Makefile
Comment (djzager) [inline Makefile:407]: This comment needs to go away.

File: Makefile
Comment (flaper87) [inline Makefile:433]: It's a small set of commands, but I'm wondering if it would be worth it to put them in a script just to avoid dealing with Make syntax.

**Review by flaper87 — COMMENTED** [review]
Looks sane so far. What would be a good testing exercise for this? I'm worried that CI won't be enough. Do you have an image generated with this PR pushed somewhere? I could give it a spin.

Is `SERVICE_ACCOUNT_NAME` already being set for the operator? Just to make sure it's passed down to the assisted-service POD.

File: Makefile
Comment (djzager) [reply Makefile:433]: Now I better understand why `hack/` directories exist :wink: . I personally prefer to keep things in the Makefile to the extent possible. That plus I only see this chunk of steps getting smaller, I think it should stay here.

**Review by carbonin — COMMENTED** [review]
I don't know enough about how operators are packaged or kustomize to fully sign off on this, but had a few small questions.

Also what are the actual differences in rbac between `ocp` and `base`? Can those be the same thing?

File: config/rbac/base/role_binding.yaml
Comment (djzager) [reply config/rbac/base/role_binding.yaml:5]: Comes back to this issue https://github.com/kubernetes-sigs/kustomize/issues/2906 

In order for the namespace to be set everywhere (of particular concern is the subject of a (cluster)rolebinding) you must have the namespace as default. I could probably set the namespace in the kustomization.yaml as `assisted-installer` to document the actual default.

Comment (carbonin) [reply config/rbac/base/role_binding.yaml:5]: Ah okay, I was confused because there is actually a "default" namespace and it seemed strange to deploy this there.

File: Makefile
Comment (rwsu) [inline Makefile:440]: The create-ocp-manifests target can be removed because the yamls in config/assisted-service have been removed.

The --apply-manifest flag can also be removed from the deploy- targets and the use of that flag can be removed from the python scripts in ./tools.

**Comment by djzager** [comment]
/hold

The only remaining thing to figure out (discovered by @flaper and @rwsu ) is that we need to figure out how we are going to build the index image going forward.

File: tools/deploy_role.py
Comment (djzager) [reply tools/deploy_role.py:20]: Done.

File: Makefile
Comment (djzager) [reply Makefile:442]: Done.

Comment (djzager) [reply Makefile:442]: Ran into a bunch of failures when I removed it and I suspect I did something wrong. Reverting that and tackling it in a follow-up PR.

**Comment by djzager** [comment]
Was able to deploy the operator in a CRC cluster using `operator-sdk run bundle` and the operator (and the assisted-service) are running with the `assisted-service` service account.

**Comment by flaper87** [comment]
/retitle OCPBUGSM-28017: Unify operator manifests and move them under config

File: docs/operator.md
Comment (flaper87) [inline docs/operator.md:109]: I think `channel` is not required (or at least we are not using it in the dev-script script): 

https://github.com/openshift-metal3/dev-scripts/blob/6c831934eb5be222eef88f2587830279bf84741d/assisted_deployment.sh#L171-L185

Do you think it's worth keeping it as part of the documentation here? Anything that we should/could mention about the use of `channel` ?

**Review by flaper87 — COMMENTED** [review]
Looks good overall, I don't want to hold this back on my comment. I will give it one more try in my env and then approve it if everything works as expected.

**Review by flaper87 — CHANGES_REQUESTED** [review]
`unable to prepare context: unable to evaluate symlinks in Dockerfile path: lstat /home/flaper87/workspace/assisted-service/deploy/olm-catalog/bundle.Dockerfile: no such file or directory`

Did you forget to add a file or am I missing something?

**Comment by rwsu** [comment]
> `unable to prepare context: unable to evaluate symlinks in Dockerfile path: lstat /home/flaper87/workspace/assisted-service/deploy/olm-catalog/bundle.Dockerfile: no such file or directory`
> 
> Did you forget to add a file or am I missing something?

bundle.Dockerfile is in the repository root. Add a symlink?

**Comment by djzager** [comment]
Tested using @flaper87's dev-scripts environment.

The operator:

```
        image: quay.io/djzager/assisted-service:config-rbac
  ...
      serviceAccount: assisted-service
      serviceAccountName: assisted-service
```

Showing everything in the namespace:

```
[dev@edge-07 dev-scripts]$ k get all -n assisted-installer
NAME                                             READY   STATUS    RESTARTS   AGE
pod/assisted-service-fbbbbc6c8-gx7zf             1/2     Running   0          52s
pod/assisted-service-operator-764c9db657-zzn6k   1/1     Running   0          115s

NAME                       TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
service/assisted-service   LoadBalancer   172.30.66.213   <pending>     8090:30785/TCP   5h31m

NAME                                        READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/assisted-service            0/1     1            0           5h31m
deployment.apps/assisted-service-operator   1/1     1            1           5h31m

NAME                                                   DESIRED   CURRENT   READY   AGE
replicaset.apps/assisted-service-7979685854            0         0         0       86s
replicaset.apps/assisted-service-8f64dd8fd             0         0         0       5h31m
replicaset.apps/assisted-service-fbbbbc6c8             1         1         0       53s
replicaset.apps/assisted-service-operator-764c9db657   1         1         1       116s
replicaset.apps/assisted-service-operator-7999c9dc64   0         0         0       2m31s

NAME                                        HOST/PORT                                                            PATH   SERVICES           PORT               TERMINATION   WILDCARD
route.route.openshift.io/assisted-service   assisted-service-assisted-installer.apps.ostest.test.metalkube.org          assisted-service   assisted-service   reencrypt     None
[dev@edge-07 dev-scripts]$ k get all -n assisted-installer
NAME                                             READY   STATUS    RESTARTS   AGE
pod/assisted-service-fbbbbc6c8-gx7zf             2/2     Running   0          103s
pod/assisted-service-operator-764c9db657-zzn6k   1/1     Running   0          2m46s

NAME                       TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
service/assisted-service   LoadBalancer   172.30.66.213   <pending>     8090:30785/TCP   5h32m

NAME                                        READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/assisted-service            1/1     1            1           5h32m
deployment.apps/assisted-service-operator   1/1     1            1           5h32m

NAME                                                   DESIRED   CURRENT   READY   AGE
replicaset.apps/assisted-service-7979685854            0         0         0       2m17s
replicaset.apps/assisted-service-8f64dd8fd             0         0         0       5h32m
replicaset.apps/assisted-service-fbbbbc6c8             1         1         1       104s
replicaset.apps/assisted-service-operator-764c9db657   1         1         1       2m47s
replicaset.apps/assisted-service-operator-7999c9dc64   0         0         0       3m22s

NAME                                        HOST/PORT                                                            PATH   SERVICES           PORT               TERMINATION   WILDCARD
route.route.openshift.io/assisted-service   assisted-service-assisted-installer.apps.ostest.test.metalkube.org          assisted-service   assisted-service   reencrypt     None
```

And the assisted-service using the `assisted-service` svcAccount created by OLM:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: assisted-service
  namespace: assisted-installer
spec:
...
    spec:
      containers:
...
        image: quay.io/ocpmetal/assisted-service:latest
        imagePullPolicy: Always
...
      serviceAccount: assisted-service
      serviceAccountName: assisted-service
```

**Review by flaper87 — APPROVED** [review]
/lgtm
:spaghetti:

**Pull request merged**
